### PR TITLE
User Activity Log - Tracking login times

### PR DESCRIPTION
**Pull request by NightHunterSV**
https://openy.atlassian.net/browse/PRODDEV-123

## Steps to test:

- [ ] Log in as an admin
- [ ] Ensure the Open Y Virtual YMCA Log module is enabled
- [ ] Navigate through the site, start the video playback, etc.
- [ ] Go to the Logs view page and ensure there are data about your activity duration
image

- [ ] Set the least value in the VY log module settings for the Activity granularity interval
image

- [ ] Be inactive during this time
- [ ] Start navigating, playing video, etc. after the interval gone
- [ ] Ensure there is another log record, gathering the activity time for the user
image

- [ ] Export logs and ensure the data about an activity are present in the resulting file
image

### CODE REVIEW

**Comment by NightHunterSV** [comment]
retest

**Review by hamrant — CHANGES_REQUESTED** [review]

**Comment by hamrant** [comment]
@NightHunterSV please increase items number on views to 50:

Screenshot from 2021-01-21 13-03-07

File: modules/openy_gc_log/src/Field/PayloadFieldItemList.php [hunk 1]
```diff
@@ -18,6 +20,30 @@ class PayloadFieldItemList extends FieldItemList {
...
+   * @param string $name
+   *   The name of the created property.
+   * @param \Drupal\Core\TypedData\TypedDataInterface $parent
+   *   The parent object of the data property.
+   *
+   * @see \Drupal\Core\TypedData\TypedDataManager::create()
+   */
+  public function __construct(DataDefinitionInterface $definition, $name, TypedDataInterface $parent) {
+    parent::__construct($definition, $name, $parent);
+    $this->dateFormatter = \Drupal::service('date.formatter');
```
Comment (anpolimus) [inline modules/openy_gc_log/src/Field/PayloadFieldItemList.php:44]: Why not to use DI here?

**Review by anpolimus — CHANGES_REQUESTED** [review]

**Comment by anpolimus** [comment]
retest

**Comment by NightHunterSV** [comment]
retest

**Comment by hamrant** [comment]
@NightHunterSV please set  `Activity granularity interval` default value to 15 mins

**Review by hamrant — CHANGES_REQUESTED** [review]
Tested, works fine. @NightHunterSV please fix route permission and we can merge this

**Review by anpolimus — APPROVED** [review]

**Review by hamrant — APPROVED** [review]

**Comment by AnastasiiaPys** [comment]
Tested, all good. Thanks.

**Pull request merged**